lib/workers/repository/reconfigure/index.ts

```diff
@@ -0,0 +1,160 @@
+import { getCache } from '../../../util/cache/repository';
+import { readLocalFile } from '../../../util/fs';
+import { getBranchCommit } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { detectConfigFile } from '../init/merge';
+import {
+  deleteReconfigureBranchCache,
+  setReconfigureBranchCache,
+} from './reconfigure-cache';
+
+export async function validateReconfigureBranch(
+  config: RenovateConfig
+): Promise<void> {
+  logger.debug('validateReconfigureBranch()');
+  const context = `renovate/config-validation`;
+
+  const branchName = `${config.branchPrefix}reconfigure`;
+  const branchExists = await scm.branchExists(branchName);
+
+  // this is something, the user initiates so skip if no branch exists
+  if (!branchExists) {
+    logger.debug('No reconfigure branch found');
+    deleteReconfigureBranchCache(); // inorder to remove cache when the branch deleted
+    return;
+  }
+
+  // look for config file
+  // 1. check reconfigure branch cache and use the conifgFileName if it exists
+  // 2. checkout reconfigure branch and look for the config file, don't assume default config fileName
+  const branchSha = getBranchCommit(branchName);
+  const cache = getCache();
+  let configFileName: string | null = null;
+  const reconfigureCache = cache.reconfigureBranchCache;
+
+  // only use cached information if it is valid
+  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
+    logger.debug('Cache is valid. Skipping validation check');
+    return;
+  }
+
+  try {
+    await scm.checkoutBranch(branchName);
+    configFileName = await detectConfigFile();
+  } catch (err) {
+    /*istanbul ignore next - should never happen*/
+    logger.error(
+      { err },
+      'Error while searching for config file in reconfigure branch'
+    );
+  }
+
+  if (!is.nonEmptyString(configFileName)) {
+    logger.warn('No config file found in reconfigure branch');
+    await platform.setBranchStatus({
+      branchName,
+      context,
+      description: 'Validation Failed - No config file found',
+      state: 'red',
+    });
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  let configFileRaw: string | null = null;
+  try {
+    configFileRaw = await readLocalFile(configFileName, 'utf8');
+  } catch (err) {
+    /*istanbul ignore next - should never happen*/
+    logger.error('Error while reading config file');
+  }
+
+  if (!is.nonEmptyString(configFileRaw)) {
+    logger.warn('Empty or invalid config file');
+    await platform.setBranchStatus({
+      branchName,
+      context,
+      description: 'Validation Failed - Empty/Invalid config file',
+      state: 'red',
+    });
+    setReconfigureBranchCache(branchSha!, configFileName, false);
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  let configFileParsed: any;
+  try {
+    const fileType = upath.extname(configFileName);
+    if (fileType === '.json') {
+      configFileParsed = JSON.parse(configFileRaw);
```

Comment:
What about the JSON5 fallback?

Reply:
That's in another PR already, should I add one here for now?

Reply:
This PR should match with what Renovate currently does today. If Renovate allows JSON5 in renovate.json then so too must this new functionality